rfcs/text/0428-extending-types.md

```diff
@@ -0,0 +1,677 @@
+
+First entity type version of `Country` without supertype:
+
+```json
+{
+  "kind": "entityType",
+  "$id": "https://blockprotocol.org/@alice/entity-type/country/v/1",
+  "type": "object",
+  "title": "Country",
+  "properties": {
+    "https://blockprotocol.org/@alice/property-type/name": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/name/v/1"
+    }
+  },
+  "required": ["https://blockprotocol.org/@alice/property-type/name"]
+}
+```
+
+Entity type `Region` with `Country` as a supertype
+
+```json
+{
+  "kind": "entityType",
+  "$id": "https://blockprotocol.org/@alice/entity-type/region/v/1",
+  "type": "object",
+  "title": "Region",
+  "allOf": [
+    { "$ref": "https://blockprotocol.org/@alice/entity-type/country/v/1" }
+  ],
+  "properties": {
+    "https://blockprotocol.org/@alice/property-type/blurb": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/blurb/v/1"
+    }
+  },
+  "required": ["https://blockprotocol.org/@alice/property-type/blurb"]
+}
+```
+
+The second, cyclic version of the `Country` entity type
+
+```json
+{
+  "kind": "entityType",
+  // Because of versioning, we can not change this version to /v/1 and create a "proper" cycle.
+  "$id": "https://blockprotocol.org/@alice/entity-type/country/v/2",
+  "type": "object",
+  "title": "Region",
+  "allOf": [
+    { "$ref": "https://blockprotocol.org/@alice/entity-type/region/v/1" }
+  ],
+  "properties": {
+    "https://blockprotocol.org/@alice/property-type/name": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/name/v/1"
+    },
+    "https://blockprotocol.org/@alice/property-type/location": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/location/v/1"
+    }
+  },
+  "required": ["https://blockprotocol.org/@alice/property-type/name"]
+}
+```
+
+This sort of type hierarchy should _not_ be accepted within the type extension system, as the circular dependencies lead to hard to reason about types. While the type hierarchy might be completely valid (as it would be in this case), we should safeguard users from making redundant type structures that look like the above.
+
+In this specific contrived example, creating a new entity type based on `Region` instead of a new version of `Country` might even encode semantic meaning better than re-defining `Country`.
+
+## Multiple supertypes - checking compatibility
+
+As described in the [Guide-level explanation](#multiple-supertypes), when extending multiple entity types, they must be able to coexist in a "compatible manner", which means that the entity types' `properties` and `links` comply with the following:
+
+- For each property (base URI on the top level of `properties`) that exists in multiple entity types:
+
+  - all entity types refer to the same versioned URI of the property **or** compatible versions of the property (through the [versioning RFC](./0408-versioning-types.md#determining-type-compatibility) definition of 'compatible')
```

Comment:
I think we should be saying property type here rather than property

Reply:
addressed in 06a981cb55010fe88c1aad80d31ba45deca85fa6